src/background/dapp-session.manager.ts

```diff
@@ -33,26 +64,30 @@ class DappSecurityContext {
     return originContentRoot === this.originContentRoot
   }
 
-  /** STORAGE FUNCTIONS */
+  public isValid(sender: chrome.runtime.MessageSender): boolean {
+    const tabId = senderTabId(sender)
+    const frameId = senderFrameId(sender)
+    const frameContentRoot = senderFrameOrigin(sender)
+    const originContentRoot = senderContentOrigin(sender)
 
-  public async setStorageItem(keyName: string, keyValue: unknown): Promise<void> {
-    const key = this.enrichStorageKey(keyName)
+    console.log(`tabid ${tabId} frameId ${frameId} frameconte ${frameContentRoot} originCon ${originContentRoot}`)
 
-    await new Promise(resolve => this.storage.set({ [key]: keyValue }, () => resolve(true)))
+    return (
+      this.isValidTabId(tabId) &&
+      this.isValidFrameId(frameId) &&
+      this.isFrameContentRoot(frameContentRoot) &&
+      this.isValidOriginContentRoot(originContentRoot)
+    )
   }
+}
 
-  public getStorageItem(keyName: string): Promise<unknown> {
-    const key = this.enrichStorageKey(keyName)
-
-    return new Promise(resolve =>
-      this.storage.get([key], result => {
-        resolve(result[key])
-      }),
-    )
+class ExtensionDappSecurityContext extends DappSecurityContext {
+  constructor() {
+    super('extension')
```

Comment:
I'm not sure this is right, because we want to make place for other extensions as well with this feature.